ScrapeEmails.java

```diff
@@ -0,0 +1,111 @@
+/*
+ ScrapeEmails.java
+
+ A command-program that takes in a domain and prints out all the email addresses found on any of 
+ the discoverable pages of the domain.
+
+ */
+
+import java.io.IOException;
+import org.jsoup.*;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import java.util.Stack;
+import java.util.HashSet;
+
+public class ScrapeEmails {
+    // Stack used to hold the remaining pages to parse
+    private Stack<String> linksToScrape; 
+    private Document doc; 
+    private String domain; 
+    // HashSet used to ensure an email is only printed once. 
+    private HashSet<String> emailSet;
+    // HashSet used to ensure a page is only parsed once. 
+    private HashSet<String> linkSet; 
+    
+    public static void main(String[] args) {
+        if(args.length == 1) { 
+            ScrapeEmails scraper = new ScrapeEmails(args[0]); 
+            scraper.findEmails();
+        }
+        else System.out.println("Please enter a domain!");
+        System.out.println("Scraping Complete");
```

Comment:
nitpick here, but this probably shouldn't be output when it didn't scrape anything (no domain input)